common/djangoapps/entitlements/models.py

```diff
@@ -0,0 +1,98 @@
+from django.db import models
+from django.contrib.auth.models import User
+from course_modes.models import CourseMode
+
+
+class CourseEntitlement(models.Model):
+    """
+    Represents a Student's Entitlement to a Course Run for a given Course.
+    """
+
+    user = models.ForeignKey(User)
+
+    # The parent courses UUID
+    course_uuid = models.UUIDField()
+
+    created = models.DateTimeField(auto_now_add=True, db_index=True)
+
+    updated = models.DateTimeField(auto_now=True)
+
+    # The date that an the entitlement expired
+    # if NULL the entitlement has not expired
+    expired_at = models.DateTimeField(null=True)
+
+    # The mode of the Course that will be applied
+    mode = models.CharField(default=CourseMode.DEFAULT_MODE_SLUG, max_length=100)
+
+    # The ID of the course enrollment for this Entitlement
+    # if NULL the entitlement is not in use
+    enrollment_course_run = models.ForeignKey('student.CourseEnrollment', null=True)
+
+    @classmethod
+    def entitlements_for_user(cls, user):
```

Comment:
Not sure why this method is necessary

Reply:
To be used in the later API code.  It was a copy and paste from the prototype.